code/messaging/src/test/java/com/adobe/marketing/mobile/MessagingTests.java

```diff
@@ -466,6 +499,136 @@ public void test_handleNotificationResponseWithEmptyAction() {
                 });
     }
 
+    @Test
+    public void test_handleNotificationResponse_withPushToInappIdentifier() {
+        Intent mockIntent = mock(Intent.class);
+        when(mockIntent.getStringExtra("messageId")).thenReturn("mockMessageId");
+        when(mockIntent.getStringExtra("adobe_xdm")).thenReturn("mockXdmData");
+        when(mockIntent.getStringExtra("adb_iam_id")).thenReturn("mockInappId");
+
+        final ArgumentCaptor<Event> eventCaptor = ArgumentCaptor.forClass(Event.class);
+        runWithMockedMobileCore(
+                eventCaptor,
+                null,
+                () -> {
+                    try (MockedStatic<Messaging> messagingMockedStatic =
+                            Mockito.mockStatic(Messaging.class, Mockito.CALLS_REAL_METHODS)) {
+                        messagingMockedStatic
+                                .when(
+                                        () ->
+                                                Messaging.updatePropositionsForSurfaces(
+                                                        anyList(), any(AdobeCallback.class)))
+                                .thenAnswer(
+                                        (Answer<Void>)
+                                                invocation -> {
+                                                    AdobeCallback<Boolean> booleanAdobeCallback =
+                                                            invocation.getArgument(1);
+                                                    booleanAdobeCallback.call(true);
+                                                    return null;
+                                                });
+
+                        Messaging.handleNotificationResponse(mockIntent, true, "mockActionId");
+
+                        Event messagingEvent = eventCaptor.getAllValues().get(1);
+                        assertEquals(
+                                "Push notification interaction event", messagingEvent.getName());
+                        assertEquals(
+                                "com.adobe.eventSource.requestContent", messagingEvent.getSource());
+                        assertEquals("com.adobe.eventType.messaging", messagingEvent.getType());
+                        assertEquals(
+                                "mockMessageId", messagingEvent.getEventData().get("messageId"));
+                        assertEquals("mockActionId", messagingEvent.getEventData().get("actionId"));
+                        assertEquals(true, messagingEvent.getEventData().get("applicationOpened"));
+                        assertEquals("mockXdmData", messagingEvent.getEventData().get("adobe_xdm"));
+                        assertEquals(
+                                "pushTracking.customAction",
```

Comment:
nit: Can we also verify that `adb_iam_id` exists in the payload?

Reply:
Discussed offline this event is still needed in case the IAM was previously cached